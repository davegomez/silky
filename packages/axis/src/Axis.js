// @flow
import React from 'react';
import type { CSS } from '../../utils/types';
import { Group } from '@silky/group';

type Props = {
  className?: string,
  css?: CSS,
  position: string,
};

export default function Text({ className = '', css, position }: Props) {
  return (
    <Group className={`silky-axis axis-${position} ${className}`} css={css}>
      {/* Test */}
    </Group>
  );
}
