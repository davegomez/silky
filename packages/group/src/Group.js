// @flow
import React from 'react';
import { getTransform } from '../../utils';
import type { CSS, Transform } from '../../utils/types';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  transform?: Transform,
};

export default function Group({
  children,
  className = '',
  css,
  transform,
  ...restProps
}: Props) {
  return (
    <g
      className={`silky-group ${className}`}
      css={css}
      transform={getTransform(transform)}
      {...restProps}
    >
      {children}
    </g>
  );
}
