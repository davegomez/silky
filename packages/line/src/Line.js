// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  x1: number | string,
  x2?: number | string,
  y1: number | string,
  y2?: number | string,
  pathLength?: number,
};

export default function Line({
  className = '',
  css,
  x1,
  x2,
  y1,
  y2,
  pathLength,
  ...restProps
}: Props) {
  return (
    <line
      className={`silky-line ${className}`}
      css={css}
      x1={x1}
      x2={x2}
      y1={y1}
      y2={y2}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
