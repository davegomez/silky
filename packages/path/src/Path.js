// @flow
import React from 'react';
import type { CSS } from '../../utils/types';

type Props = {
  className?: string,
  css?: CSS,
  d: string,
  pathLength?: number,
};

export default function Path({
  className = '',
  css,
  d,
  pathLength,
  ...restProps
}: Props) {
  return (
    <path
      className={`silky-path ${className}`}
      css={css}
      d={d}
      pathLength={pathLength}
      {...restProps}
    />
  );
}
