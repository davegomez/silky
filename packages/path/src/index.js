export { default as Path } from './Path';
