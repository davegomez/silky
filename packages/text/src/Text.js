// @flow
import React from 'react';

type CSS = {
  map: string,
  name: string,
  styles: string,
};

type Adjust = 'spacing' | 'spacingAndGlyphs';

type Props = {
  children: any,
  className?: string,
  css?: CSS,
  dx?: number | string,
  dy?: number | string,
  lengthAdjust?: Adjust,
  transform?: {
    left?: number,
    top?: number,
    rotation?: number | string,
  },
  textLength?: number | string,
  x?: number | string,
  y?: number | string,
};

export default function Text({
  children,
  className = '',
  css,
  dx,
  dy,
  lengthAdjust,
  transform = {},
  textLength,
  x,
  y,
  ...restProps
}: Props) {
  var { left, rotation, top } = transform;
  var translate = left && top ? `translate(${left}, ${top})` : '';
  var rotate = rotation ? `rotate(${rotation})` : '';

  return (
    <text
      className={`silky-text ${className}`}
      css={css}
      dx={dx}
      dy={dy}
      lengthAdjust={lengthAdjust}
      textLength={textLength}
      transform={`${translate} ${rotate}`}
      x={x}
      y={y}
      {...restProps}
    >
      {children}
    </text>
  );
}
