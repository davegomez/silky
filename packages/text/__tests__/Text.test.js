import React from 'react';
import serializer, { matchers } from 'jest-emotion';
import { render, cleanup } from 'react-testing-library';
import { css } from '@emotion/core';
import { Text } from '../src';

expect.addSnapshotSerializer(serializer);
expect.extend(matchers);
afterEach(cleanup);

test('Text', () => {
  const { container } = render(
    <svg>
      <Text>foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.innerHTML).toBe('foo');
});

test('Text with custom class', () => {
  const { container } = render(
    <svg>
      <Text className="foo">foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.classList.contains('foo')).toBe(true);
  expect(text.innerHTML).toBe('foo');
});

test('Text with Emotion', () => {
  const styles = css`
    fill: black;
  `;
  const { container } = render(
    <svg>
      <Text css={styles}>foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text).toHaveStyleRule('fill', 'black');
  expect(text.innerHTML).toBe('foo');
});

test('Text with horizontal and vertical text shift', () => {
  const { container } = render(
    <svg>
      <Text dx={10} dy={10}>
        foo
      </Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.getAttribute('dx')).toBe('10');
  expect(text.getAttribute('dy')).toBe('10');
  expect(text.innerHTML).toBe('foo');
});

test('Text with lengthAdjust:spacing', () => {
  const { container } = render(
    <svg>
      <Text lengthAdjust="spacing">foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.getAttribute('lengthAdjust')).toBe('spacing');
  expect(text.innerHTML).toBe('foo');
});

test('Text with lengthAdjust:spacingAndGlyphs', () => {
  const { container } = render(
    <svg>
      <Text lengthAdjust="spacingAndGlyphs">foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.getAttribute('lengthAdjust')).toBe('spacingAndGlyphs');
  expect(text.innerHTML).toBe('foo');
});

test('Text with textLength', () => {
  const { container } = render(
    <svg>
      <Text textLength="10%">foo</Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.getAttribute('textLength')).toBe('10%');
  expect(text.innerHTML).toBe('foo');
});

test('Text with x and y coordinates', () => {
  const { container } = render(
    <svg>
      <Text x={10} y={10}>
        foo
      </Text>
    </svg>
  );

  const text = container.querySelector('text');
  expect(text).toMatchSnapshot();
  expect(text.classList.contains('silky-text')).toBe(true);
  expect(text.getAttribute('x')).toBe('10');
  expect(text.getAttribute('y')).toBe('10');
  expect(text.innerHTML).toBe('foo');
});
