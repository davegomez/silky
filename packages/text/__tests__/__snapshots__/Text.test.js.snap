// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text 1`] = `
<text
  class="silky-text "
  transform=" "
>
  foo
</text>
`;

exports[`Text width transform rotate 1`] = `
<text
  class="silky-text "
  transform=" rotate(10 20, 30)"
>
  foo
</text>
`;

exports[`Text width transform translate 1`] = `
<text
  class="silky-text "
  transform="translate(10, 10) "
>
  foo
</text>
`;

exports[`Text width transform translate and rotate 1`] = `
<text
  class="silky-text "
  transform="translate(10, 10) rotate(90)"
>
  foo
</text>
`;

exports[`Text with Emotion 1`] = `
.emotion-0 {
  fill: black;
}

<text
  class="silky-text emotion-0"
  transform=" "
>
  foo
</text>
`;

exports[`Text with custom class 1`] = `
<text
  class="silky-text foo"
  transform=" "
>
  foo
</text>
`;

exports[`Text with horizontal and vertical text shift 1`] = `
<text
  class="silky-text "
  dx="10"
  dy="10"
  transform=" "
>
  foo
</text>
`;

exports[`Text with lengthAdjust:spacing 1`] = `
<text
  class="silky-text "
  lengthAdjust="spacing"
  transform=" "
>
  foo
</text>
`;

exports[`Text with lengthAdjust:spacingAndGlyphs 1`] = `
<text
  class="silky-text "
  lengthAdjust="spacingAndGlyphs"
  transform=" "
>
  foo
</text>
`;

exports[`Text with textLength 1`] = `
<text
  class="silky-text "
  textLength="10%"
  transform=" "
>
  foo
</text>
`;

exports[`Text with x and y coordinates 1`] = `
<text
  class="silky-text "
  transform=" "
  x="10"
  y="10"
>
  foo
</text>
`;
